azblob/url_container.go

```diff
@@ -241,7 +238,7 @@ func (c ContainerURL) ListBlobsFlatSegment(ctx context.Context, marker Marker, o
 // For more information, see https://docs.microsoft.com/rest/api/storageservices/list-blobs.
 func (c ContainerURL) ListBlobsHierarchySegment(ctx context.Context, marker Marker, delimiter string, o ListBlobsSegmentOptions) (*ListBlobsHierarchySegmentResponse, error) {
 	if o.Details.Snapshots {
-		panic("snapshots are not supported in this listing operation")
+		sanityCheckFailed("snapshots are not supported in this listing operation")
```

Comment:
This function already has an error on the return signature. Why not return an error here rather than a panic?

Reply:
The discussion will continue [here](https://github.com/Azure/azure-storage-blob-go/pull/73#discussion_r222069003).

Reply:
The discussion will continue [here](https://github.com/Azure/azure-storage-blob-go/pull/73#discussion_r222069003).